packages/contracts/test/core/acl/integration-tests/parameter-scoping-oracle.ts

```diff
@@ -0,0 +1,96 @@
+import chai, {expect} from 'chai';
+import {ethers} from 'hardhat';
+import chaiUtils from '../../../test-utils';
+import {customError} from '../../../test-utils/custom-error-helper';
+import {SignerWithAddress} from '@nomiclabs/hardhat-ethers/signers';
+import {
+  TestParameterScopingACLOracle,
+  TestComponent,
+  DAO,
+} from '../../../../typechain';
+
+chai.use(chaiUtils);
+
+const DO_SOMETHING_ROLE = ethers.utils.id('DO_SOMETHING_ROLE');
+
+describe('TestParameterScopingOracle', function () {
+  let signers: SignerWithAddress[];
+  let parameterOracle: TestParameterScopingACLOracle;
+  let testComponent: TestComponent;
+  let dao: DAO;
+  let ownerAddress: string;
+
+  before(async () => {
+    signers = await ethers.getSigners();
+    ownerAddress = await signers[0].getAddress();
+
+    // create a DAO
+    const DAO = await ethers.getContractFactory('DAO');
+    dao = await DAO.deploy();
+    await dao.initialize('0x', ownerAddress, ethers.constants.AddressZero);
+
+    // create a component
+    const TestComponent = await ethers.getContractFactory('TestComponent');
+    testComponent = await TestComponent.deploy();
+    await testComponent.initialize(dao.address);
+
+    // create parameter oracle
+    const ParameterOracle = await ethers.getContractFactory(
+      'TestParameterScopingACLOracle'
+    );
+    parameterOracle = await ParameterOracle.deploy();
+
+    // Give signers[0] the DO_SOMETHING_ROLE on the TestComponent
+    dao.grantWithOracle(
+      testComponent.address,
+      ownerAddress,
+      DO_SOMETHING_ROLE,
+      parameterOracle.address
+    );
+  });
+
+  describe('oracle conditions:', async () => {
+    it('adds if the first parameter is larger than the second', async () => {
+      let param1 = 10;
+      let param2 = 1;
+
+      expect(
+        await testComponent.callStatic.addPermissioned(param1, param2)
+      ).to.be.equal(11);
+    });
+
+    it('reverts if the oracle is called by the wrong function', async () => {
+      let param1 = 10;
+      let param2 = 1;
+
+      await expect(
+        testComponent.callStatic.subPermissioned(param1, param2)
+      ).to.be.revertedWith(
+        customError(
+          'ACLAuth',
+          testComponent.address,
+          testComponent.address,
+          ownerAddress,
+          DO_SOMETHING_ROLE
+        )
+      );
+    });
+
+    it('reverts if the first parameter is not larger than the second', async () => {
+      let param1 = 1;
+      let param2 = 1;
+
+      await expect(
+        testComponent.callStatic.addPermissioned(param1, param2)
+      ).to.be.revertedWith(
+        customError(
+          'ACLAuth',
+          testComponent.address,
+          testComponent.address,
+          ownerAddress,
+          DO_SOMETHING_ROLE
+        )
+      );
+    });
```

Comment:
Are there further ways we can test again unintended usages?

`Component` is the most fundamental piece of solidity we have, all attempts to put it under heavy pressure and check that it performs are very welcome.
Maybe Sarkawt and @mathewmeconry can also help in this sense

Reply:
we could use echidna and do fuzzing tests to make it a little more bulletproof

Reply:
Yes, we should definitely do this ([see our Jira issue](https://aragonassociation.atlassian.net/browse/APP-50)).

Reply:
I see, but fuzzy testing feels like a tool for a different job. What I was referring to was more about intentional tests trying to find tricky usages, more than expecting fuzzed data to do the job.

Additionally, fuzzy tests ran on `callStatic` calls may not "return" any incorrect values but effectively manage to alter the on-chain state (without the test realizing it).

Reply:
ping

Reply:
Well, given that this is a "testing" PR (no feature), I don't see why we should wait for `APP-51` to add even more tests. It will only get bigger, and realizing what is pending test is only going to become more challenging. Now it's fresh in our minds and this review is already 15 days old. 

If `Component` is out of the scope of this PR, I understand, but then let's at least enumerate all the potentially missing cases to test, and add them explicitly to `APP-51` so we don't forget.